src/api.js

```diff
@@ -118,25 +118,40 @@ async function setNamespace(context, namespace, contact) {
   try {
     newNamespace = await context.Namespaces.create({
       namespace:  namespace,
-      username:   slugid.v4(),
+      username:   namespace,
       password:   slugid.v4(),
       created:    new Date(),
       expires:    taskcluster.fromNow('1 day'),
+      rotationState:  '1',
+      nextRotation: taskcluster.fromNow('1 hour'),
       contact:    contact,
     });
     
+    await context.rabbit.createUser(namespace.concat('-1'), newNamespace.password, ['taskcluster-pulse']);
+    await context.rabbit.createUser(namespace.concat('-2'), newNamespace.password, ['taskcluster-pulse']);
+    
+    //set up user pairs in rabbitmq, both users are used for auth rotations
     await context.rabbit.setUserPermissions(
-      user = await context.rabbit.createUser(newNamespace.username, newNamespace.password, ['taskcluster-pulse']),
-      vhost             = '/',
-      configurePattern  = '',
-      writePattern      = `taskcluster/(exchanges|queues)/${newNamespace.namespace}/.*`,
-      readPattern       = 'taskcluster/exchanges/.*',
+      namespace.concat('-1'),                                         //username
+      '/',                                                            //vhost
+      `taskcluster/(exchanges|queues)/${newNamespace.namespace}/.*`,  //configure pattern
+      `taskcluster/(exchanges|queues)/${newNamespace.namespace}/.*`,  //write pattern
+      'taskcluster/exchanges/.*'                                      //read pattern
```

Comment:
Sorry I was implying that we missed the read permission on __queues__ (of the user's scope). Currently this `readPattern` only grants access to exchanges (`bind` semantic), not queues (`consume` semantic). Read permission on queues is needed so that users can actually consume from the queue they created (which is bind to some arbitrary exchange).

https://www.rabbitmq.com/access-control.html